iot-hub/Samples/device/DeviceReconnectionSample/DeviceReconnectionSample.cs

```diff
@@ -98,101 +116,96 @@ private void ConnectionStatusChangeHandler(ConnectionStatus status, ConnectionSt
-                }
-                catch (Exception ex) when (ex is IotHubException exception && exception.IsTransient)
-                {
-                    // Inspect the exception to figure out if operation should be retried, or if user-input is required.
-                    _logger.LogError($"An IotHubException was caught, but will try to recover and retry explicitly: {ex}");
-                }
-                catch (Exception ex) when (ExceptionHelper.IsNetworkExceptionChain(ex))
+                // If the client is in Connected state, perform the desired operation.
+                if (s_connectionStatus == ConnectionStatus.Connected)
                 {
-                    _logger.LogError($"A network related exception was caught, but will try to recover and retry explicitly: {ex}");
-                }
+                    count++;
+                    _logger.LogInformation($"Device sending message {count} to IoTHub...");
+
+                    try
+                    {
+                        await SendMessageAsync(count);
+                    }
+                    catch (Exception ex) when (ex is IotHubException exception && exception.IsTransient)
+                    {
+                        // Inspect the exception to figure out if operation should be retried, or if user-input is required.
+                        _logger.LogError($"An IotHubException was caught, but will try to recover and retry explicitly: {ex}");
+                    }
+                    catch (Exception ex) when (ExceptionHelper.IsNetworkExceptionChain(ex))
+                    {
+                        _logger.LogError($"A network related exception was caught, but will try to recover and retry explicitly: {ex}");
+                    }
 
-                await Task.Delay(s_sleepDuration);
+                    await Task.Delay(s_sleepDuration);
+                }
             }
         }
 
         private async Task ReceiveMessagesAsync(CancellationToken cancellationToken)
         {
             while (!cancellationToken.IsCancellationRequested)
             {
-                _logger.LogInformation($"Device waiting for C2D messages from the hub - for {s_sleepDuration}...");
-                _logger.LogInformation("Use the IoT Hub Azure Portal or Azure IoT Explorer to send a message to this device.");
-
-                try
-                {
-                    await ReceiveMessageAndCompleteAsync();
-                }
-                catch (Exception ex) when (ex is DeviceMessageLockLostException)
-                {
-                    _logger.LogWarning($"Attempted to complete a received message whose lock token has expired; ignoring: {ex}");
-                }
-                catch (Exception ex) when (ex is IotHubException exception && exception.IsTransient)
+                // If the client is in Connected state, perform the desired operation.
+                if (s_connectionStatus == ConnectionStatus.Connected)
```

Comment:
Feel like there should be a property or method to abstract this notion away. `IsConnected` or `CanSend()` or something like that.